common/src/main/java/com/wynntils/functions/CharacterFunctions.java

```diff
@@ -87,6 +87,39 @@ public Double getValue(FunctionArguments arguments) {
         }
     }
 
+    public static class StatusEffectsFunction extends Function<String> {
+        @Override
+        public String getValue(FunctionArguments arguments) {
+            List<String> statusEffectsList = Models.StatusEffect.getStatusEffects().stream()
+                    .map(statusEffect -> statusEffect.asString().getString())
+                    .toList();
+
+            return String.join("\n", statusEffectsList);
+        }
+    }
+
+    public static class ContainsStatusEffectFunction extends Function<Boolean> {
+        @Override
+        public Boolean getValue(FunctionArguments arguments) {
+            String query = arguments.getArgument("query").getStringValue();
+            return !Models.StatusEffect.getStatusEffects().stream()
+                    .filter(statusEffect -> statusEffect.asString().getString().contains(query))
```

Comment:
Isn't it more logical to match the whole string? Since we are talking about a list of effects, we want to do contains on the list, not the string.

Reply:
The issue is each effect contains an icon, cooldown, many parts that are irrelevant and hard to keep track of when you are writing info boxes, it's a lot easier to just match part of the string for the purposes of using it

Reply:
And there is no method for getting only the name part as a string for status effect? Maybe there should be one then.

Reply:
The line looks like this: *ICON* Radiant (00:15)
After every second, it will change to lets say *ICON* Radiant (00:14)

I need the function to simply match if the word Radiant is present, it would be a lot harder if it was trying to match the entire effect name rather than just part of it

It sure is a bit weird since status_effect_active("a") would most likely return True if any effect had the letter "a" in them, but... how complicated do we want to make this

Reply:
> And there is no method for getting only the name part as a string for status effect? Maybe there should be one then.

I could work on that

Reply:
> > And there is no method for getting only the name part as a string for status effect? Maybe there should be one then.
> 
> 
> 
> I could work on that

Afaik and remember, the StatusEffect class holds everything parsed nicely, so it's only a getter that you need. If it is not parsed there, don't spend time trying to parse it correctly.

Reply:
> > > And there is no method for getting only the name part as a string for status effect? Maybe there should be one then.
> > 
> > 
> > I could work on that
> 
> Afaik and remember, the StatusEffect class holds everything parsed nicely, so it's only a getter that you need. If it is not parsed there, don't spend time trying to parse it correctly.

Apparently, it is. Thank you. I'll do some testing as to what exactly it holds, I'll just use that if it's good.